### PR TITLE
ORCA falls back on ONLY keyword in FROM clause

### PR DESCRIPTION
**Pull request by sambitesh**
Signed-off-by: Sambitesh Dash <sdash@pivotal.io>

### CODE REVIEW

File: src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp [hunk 1]
```diff
@@ -2984,6 +2984,11 @@ CTranslatorQueryToDXL::PdxlnFromRelation
 	ULONG //ulCurrQueryLevel 
 	)
 {
+	if (false == prte->inh)
```
Comment (vraghavan78) [inline src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp]: what if you have a regular table and you use ONLY. It will still fallback.

Comment (dhanashreek89) [reply src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp]: how does `prte->inh` implies that there is ONLY keyword in the FROM clause? `prte->inh` just represents if the relation is inherited or not, right?

**Comment by sambitesh** [comment]
@dhanashreek89 @vraghavan78 we see that a comment is in order. Other than that are we good to merge this?

Jesse and Sam

**Comment by mkiyama** [comment]
Will this work after the fix? Currently it doesn't in a master branch build.

create table b1 (bc1 text, bc2 text) distributed randomly ;
create table a1 (ac1 text, ac2 text) INHERITS b1 distributed randomly ;
insert into a1 values ('a', 'b', 'c', 'd'), ('aa', 'ba', 'ca', 'da') ;

I was expecting only 2 columns with optimizer='off'. Am I missing something?

select * from only a1 ;
bc1 | bc2 | ac1 | ac2
-----+-----+-----+-----
 a   | b   | c   | d
 aa  | ba  | ca  | da
(2 rows)

**Comment by d** [comment]
table `a1` has four columns, what did you expect?

**Comment by mkiyama** [comment]
Do you have a simple example that uses INHERIT where the ONLY keyword changes the results returned or are those two keywords unrelated?

I am looking for an example that is not a partitioned table.

thanks

**Pull request merged**